prow/git/git.go

```diff
@@ -316,6 +334,17 @@ func (r *Repo) Config(key, value string) error {
 	return nil
 }
 
+// GitRemote builds a remote url from user, pasword, and a slice of path items
+func GitRemote(base string, user string, pass string, repo string) (*url.URL, error) {
```

Comment:
Why do you want to export this method? If you don't want to (don't think its needed), please don't capitalize the method.

Reply:
I am exporting this so we can test it, `git_test.go` is a different package then `git.go`, so we need to export functions to test them... not sure why they are that way though 🤷‍♂️

Reply:
These two files are in the same package (git)

Reply:
Huh, well they both ought to be in the same package. Not sure why the test is in git_test rather than git.

Reply:
Huh, well they both ought to be in the same package. Not sure why the test is in git_test rather than git.

Reply:
I think this is because of the way local_git is implemented. Can we leave this git/git_test re-work off this PR? It seems like it should have it's own re-work PR.